### PR TITLE
quizzes api generation

### PR DESCRIPTION
**Pull request by Adam3004**

### CODE REVIEW

File: src/main/resources/api.yaml [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Questions of a single Quiz
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Question'
```
Comment (krisx303) [inline src/main/resources/api.yaml]: Probably here should be returned list of questions

File: src/main/resources/api.yaml [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+              items:
+                $ref: '#/components/schemas/Question'
+      responses:
+        '201':
+          description: Questions created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Question'
+  /questions/{quizId]:
```
Comment (krisx303) [inline src/main/resources/api.yaml:192]: Maybe /quizzes/quizId/questions?

File: src/main/resources/api.yaml [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        correct:
+          type: integer
+          format: int32
+    QuizDetails:
+      type: object
+      required:
+        - id
+        - name
+        - description
+        - numberOfExercises
```
Comment (krisx303) [inline src/main/resources/api.yaml]: Number of questions?

File: src/main/resources/api.yaml [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        - lastScore
```
Comment (krisx303) [inline src/main/resources/api.yaml:264]: I just realised we need to store scores on backend xD - on start we can just add simple endpoint with correct/incorrect numbers in request body, but in future we probably would like to store more details - to generate  attempt dedicated for user

File: src/main/resources/api.yaml [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A single quiz
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Quiz'
+  /quizzes/details/{quizId}:
```
Comment (krisx303) [inline src/main/resources/api.yaml]: Why not /quizzes/quizId/details?

File: src/main/resources/api.yaml [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Details of a single Quiz
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/QuizDetails'
+  /quizzes/details:
```
Comment (krisx303) [inline src/main/resources/api.yaml]: It should have more sense for me if we provide here a quizId - we have an endpoint to create base quiz

Comment (Adam3004) [reply src/main/resources/api.yaml]: you mean /quizzes/{quizId}/details?

File: src/main/resources/api.yaml (hunk 1 above)
Comment (Adam3004) [reply src/main/resources/api.yaml]: yup

File: src/main/resources/api.yaml (hunk 3 above)
Comment (Adam3004) [reply src/main/resources/api.yaml]: i took names from frontend but it can be changed

File: src/main/resources/api.yaml (hunk 4 above)
Comment (Adam3004) [reply src/main/resources/api.yaml:264]: sure

File: src/main/resources/api.yaml (hunk 5 above)
Comment (Adam3004) [reply src/main/resources/api.yaml]: nice question

**Review by krisx303 — APPROVED** [review]

**Pull request merged**